books_to_scrape/spiders/books.py

```diff
@@ -0,0 +1,46 @@
+import scrapy
+from scrapy.http import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+
+    def parse(self, response: Response, **kwargs):
+        book_urls = response.css(".image_container > a::attr(href)").getall()
+        for book_url in book_urls:
+            book_url = response.urljoin(book_url)
+            yield response.follow(book_url, callback=self.parse_book)
+
+        next_page = response.css(".pager .next a::attr(href)").get()
+        if next_page:
+            next_page_url = response.urljoin(next_page)
+            yield response.follow(next_page_url, callback=self.parse)
+
+
+    def parse_book(self, response: Response):
+        rating_map = {"One": 1, "Two": 2, "Three": 3, "Four": 4, "Five": 5}
+
+        yield {
+            "title": response.css(".product_page h1::text").get(),
+            "price": float(
+                response.css(".price_color::text").get().replace("£", "")
+            ),
+            "amount_in_stock": response.css(
+                ".instock.availability::text"
+            ).re_first(r"\((\d+) available\)"),
```

Comment:
The `amount_in_stock` extraction uses a regex to find the number of available items. Ensure that the regex correctly matches the format of the availability text. If the format varies, consider adding error handling or additional checks to avoid potential issues.